### PR TITLE
[ACR] BREAKING CHANGE: `az acr login --expose-token` does not accept username and password

### LINKED ISSUE #13847: `az acr login --expose-token` should throw error if username or password is specified
**Describe the bug**
When a username or password is specified, the `az acr login --expose-token` returns a invalid result.

**To Reproduce**
`az acr login -n myregistry -u test -p test --expose-token`

image

**Expected behavior**
Throw error and output a warning if username or password is specified, as exposing token for ACR admin users and token users makes no sense.

**Environment summary**

azure-cli 2.3.0

**Additional context**

### PR DESCRIPTION
**Pull request by Wwwsylvia**
**Description**  

Fix #13847
BREAKING CHANGE: The `az acr login --expose-token` now raises error if username or password is specified.

**Testing Guide**  

`az acr login -n myregistry -u username -p password --expose-token`

Before:
image

After:
image

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

**Comment by Wwwsylvia** [comment]
@shizhMSFT @yugangw-msft  Please help to review, thanks

File: src/azure-cli/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -185,6 +185,9 @@ def acr_login(cmd,
               password=None,
               expose_token=False):
     if expose_token:
+        if username or password:
+            raise CLIError("Exposing token for admin users and token users is unsupported.")
```
Comment (shizhMSFT) [inline src/azure-cli/azure/cli/command_modules/acr/custom.py:189]: Customer will be confused on the following output
```
$ az acr login --expose-token --password hello
Exposing token for admin users and token users is unsupported.
```

**Comment by Wwwsylvia** [comment]
@fengzhou-msft @qwordy This PR is good to merge if no concern

**Pull request merged**